cpp/src/unary/cast_ops.cu

```diff
@@ -172,7 +172,9 @@ struct CastTimestampTo_Dispatcher {
         std::is_same<TypeTo, cudf::timestamp>::value,
     void>
     operator()(gdf_column const* input, gdf_column *output) {
-        if( input->dtype_info.time_unit == TIME_UNIT_s && output->dtype_info.time_unit == TIME_UNIT_ms ) 
+        if( input->dtype_info.time_unit == output->dtype_info.time_unit ) 
+            cudf::unary::Launcher<TypeFrom, TypeTo, DeviceCast<TypeFrom, TypeTo> >::launch(input, output);
+        else if( input->dtype_info.time_unit == TIME_UNIT_s && output->dtype_info.time_unit == TIME_UNIT_ms ) 
```

Comment:
This is an ugly if/else... I guess it exists because the METRIC_FACTOR* is a template parameter. Does it need to be a template parameter? What are we gaining by making it one? @devavret ? Sure would be nicer to either compute the conversion factor from the inputs on the fly or to store it in a lookup table...  Why isn't that possible?

Reply:
I can't speak for @devavret, but it looks like it's reusing the same `unary::Launcher` and `gpu_op_kernel` as the other unaryops.

Reply:
Spoke to @devavret and it's an existing flaw... We should fix it, but probably not in this PR.

Reply:
Adding it here: @trxcllnt is correct. Unary launcher didn't take extra args. Timestamps are due for an overhaul in libcudf and the plan is to make each resolution its own type. This part of the code will then be redundant.